src/main/java/com/epam/edumanagementsystem/student/rest/api/StudentController.java

```diff
@@ -67,27 +65,49 @@ public String createStudentSection(@ModelAttribute("student") @Valid StudentDto
         model.addAttribute("genders", Gender.values());
         model.addAttribute("parents", ParentMapper.toParentListWithoutSaveUser(findAllParents()));
         model.addAttribute("classes", findAllClasses());
-        if (userService.checkDuplicationOfEmail(studentDto.getEmail())) {
-            model.addAttribute("duplicated", "A user with the specified email already exists");
-            return STUDENT_HTML;
-        }
-        if (result.hasErrors()) {
-            if (!result.hasFieldErrors("email")) {
-                if (!validateEmail(studentDto.getEmail())) {
-                    model.addAttribute("invalid", "Email is invalid");
-                    return STUDENT_HTML;
-                }
-            }
-            return STUDENT_HTML;
-        } else if (!validateEmail(studentDto.getEmail())) {
-            model.addAttribute("invalid", "Email is invalid");
+        userService.checkDuplicationOfEmail(studentDto.getEmail(), model);
+        PasswordValidation.validatePassword(studentDto.getPassword(), model);
+        EmailValidation.validate(studentDto.getEmail(), model);
+
+        if (result.hasErrors() || model.containsAttribute("blank")
+                || model.containsAttribute("invalidPassword")
+                || model.containsAttribute("invalidEmail")) {
             return STUDENT_HTML;
         }
         studentDto.setPassword(bcryptPasswordEncoder.encode(studentDto.getPassword()));
         studentService.create(studentDto);
         return "redirect:/students";
     }
 
+    @GetMapping("/{id}/profile")
+    public String openStudentProfile(@PathVariable("id") Long studentId, Model model) {
+        model.addAttribute("existingStudent", studentService.findByStudentId(studentId));
+        model.addAttribute("bloodGroups", BloodGroup.values());
+        model.addAttribute("genders", Gender.values());
+        model.addAttribute("parents", findAllParents());
+        model.addAttribute("classes", findAllClasses());
+        return "studentProfile";
+    }
+
+    @PostMapping("/{id}/profile")
+    public String editStudentPersonalInformation(@ModelAttribute("existingStudent") @Valid StudentDto updatableStudent,
+                                                 BindingResult result, @PathVariable("id") Long studentId, Model model) {
+        model.addAttribute("bloodGroups", BloodGroup.values());
+        model.addAttribute("genders", Gender.values());
+        model.addAttribute("parents", findAllParents());
+        model.addAttribute("classes", findAllClasses());
+        if (!updatableStudent.getEmail().equals(studentService.findByStudentId(studentId).getEmail())) {
+            userService.checkDuplicationOfEmail(updatableStudent.getEmail(), model);
+        }
+        EmailValidation.validate(updatableStudent.getEmail(), model);
+
+        if (result.hasErrors() || model.containsAttribute("invalidEmail") || model.containsAttribute("duplicated")) {
+            return "studentProfile";
```

Comment:
I think you need to add the student's name and surname to the model's attribute to escape dynamically changing student's name and surname bug on the top of the popup page. Also make the corresponding change in sutdentProfile html.